File: frontend/src/types/session.ts
export interface LanguageOption {
  code: string
  name: string
  native_name: string
  is_default: boolean
  is_active: boolean
}

export interface MedicalStream {
  id: string
  name: string
  code: string
  description?: string | null
  is_active: boolean
  created_at: string
}

export interface Department {
  id: string
  hospital_id: string
  name: string
  code: string
  description?: string | null
  stream_code?: string | null
  is_active: boolean
  created_at: string
}

export interface ConsentRecord {
  id: string
  session_id: string
  patient_id: string
  consent_type: string
  consent_text: string
  language: string
  is_granted: boolean
  consented_at?: string | null
  withdrawn_at?: string | null
  created_at: string
}

export interface IntakeSession {
  id: string
  patient_id: string
  hospital_id: string
  medical_stream_id?: string | null
  department_id?: string | null
  language: string
  status: string
  started_at?: string | null
  completed_at?: string | null
  created_at: string
  updated_at: string
  patient?: {
    id: string
    patient_code: string
    full_name: string
  } | null
  medical_stream?: MedicalStream | null
  department?: Department | null
  consents?: ConsentRecord[]
}

export interface SessionCreatePayload {
  patient_id: string
  hospital_id?: string | null
  medical_stream_id?: string | null
  department_id?: string | null
  language?: string
}

export interface SessionUpdatePayload {
  medical_stream_id?: string | null
  department_id?: string | null
  language?: string | null
  status?: string | null
}

export interface ConsentSubmitPayload {
  patient_id: string
  consent_type: string
  consent_text: string
  language: string
  is_granted: boolean
}

File: backend/app/schemas/session.py
"""IntakeSession and Consent Pydantic schemas."""
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field

from app.schemas.department import DepartmentRead
from app.schemas.medical_stream import MedicalStreamRead
from app.schemas.patient import PatientRead


class ConsentCreate(BaseModel):
    patient_id: uuid.UUID
    consent_type: str = Field(default="CLINICAL_INTAKE", max_length=50)
    consent_text: str
    language: str = Field(default="en", max_length=10)
    is_granted: bool


class ConsentRead(BaseModel):
    id: uuid.UUID
    session_id: uuid.UUID
    patient_id: uuid.UUID
    consent_type: str
    consent_text: str
    language: str
    is_granted: bool
    consented_at: Optional[datetime] = None
    withdrawn_at: Optional[datetime] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SessionCreate(BaseModel):
    patient_id: uuid.UUID
    hospital_id: Optional[uuid.UUID] = None
    medical_stream_id: Optional[uuid.UUID] = None
    department_id: Optional[uuid.UUID] = None
    language: str = Field(default="en", max_length=10)


class SessionUpdate(BaseModel):
    medical_stream_id: Optional[uuid.UUID] = None
    department_id: Optional[uuid.UUID] = None
    language: Optional[str] = Field(default=None, max_length=10)
    status: Optional[str] = Field(default=None, max_length=50)


class SessionRead(BaseModel):
    id: uuid.UUID
    patient_id: uuid.UUID
    hospital_id: uuid.UUID
    medical_stream_id: Optional[uuid.UUID] = None
    department_id: Optional[uuid.UUID] = None
    language: str
    status: str
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    patient: Optional[PatientRead] = None
    medical_stream: Optional[MedicalStreamRead] = None
    department: Optional[DepartmentRead] = None
    consents: list[ConsentRead] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)
